mlir/include/mlir/Interfaces/ControlFlowInterfaces.td

```diff
@@ -343,6 +343,27 @@ def RegionBranchTerminatorOpInterface :
   }];
 }
 
+def SelectLikeOpInterface : OpInterface<"SelectLikeOpInterface"> {
+  let description = [{
+    This interface provides information for select-like operations, i.e.,
+    operations that forward specific operands to the output, depending on a
```

Comment:
This needs a bit more documentation in terms of the invariant and promises associated with the interface I believe, in particular in terms of transforms: can we always fold or propagate based on the only behavior of a regular select?

Reply:
+1, I'm curious about poison propagation

Reply:
Attempted to address this by restricting the semantics to fit LLVM's and arith dialect's select instruction/operation.